tools/resolveAgencyDuplicates.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+require(dirname(__FILE__) . '/bootstrap.php');
+
+class resolveAgencyDuplicates extends \PKP\cliTool\ResolveAgencyDuplicatesTool
+{
+    /** @inheritdoc */
+    protected array $agencies = [
+        'crossref' => [
+            'status' => 'crossref::status',
+            'additionalFields' => [
+                'crossref::registeredDoi',
+                'crossref::batchId',
+                'crossref::failedMsg',
+            ],
+        ],
+        'datacite' => [
+            'status' => 'datacite::status',
+            'additionalFields' => [
+                'datacite::registeredDoi',
+            ],
+        ],
+        'medra' => [
+            'status' => 'medra::status',
+            'additionalFields' => [
+                'medra::registeredDoi',
+            ],
+        ],
+    ];
+
+    protected function handleResolution(array $agencyFields): void
+    {
+        $submissionIds = $this->getSubmissionIds();
+        $galleyIds = $this->getGalleyIds();
+        $issueIds = $this->getIssueIds();
+
+        DB::table('submission_settings')
+            ->whereIn('setting_name', $agencyFields)
+            ->whereIn('submission_id', $submissionIds)
+            ->delete();
+        $this->print("Removed duplicates for {$submissionIds->count()} submission(s)");
+
+        DB::table('publication_galley_settings')
+            ->whereIn('setting_name', $agencyFields)
+            ->whereIn('galley_id', $galleyIds)
+            ->delete();
+        $this->print("Removed duplicates for {$galleyIds->count()} galley(s)");
+
+        DB::table('issue_settings')
+            ->whereIn('setting_name', $agencyFields)
+            ->whereIn('issue_id', $issueIds)
+            ->delete();
+        $this->print("Removed duplicates for {$issueIds->count()} issues(s)");
+    }
+
+    protected function test(): void
+    {
+        $this->print('IDs with duplicate DOI registration metadata:');
+        $this->print("Submissions: {$this->getSubmissionIds()}");
+        $this->print("Galleys: {$this->getGalleyIds()}");
+        $this->print("Issues: {$this->getIssueIds()}");
+    }
+
+    private function getSubmissionIds(): Collection
+    {
+        return DB::table('submission_settings')
+            ->whereIn('setting_name', array_map(fn ($item) => $item['status'], $this->agencies))
+            ->groupBy('submission_id')
+            ->havingRaw('COUNT(submission_id) > 1')
+            ->select(['submission_id'])
+            ->get()
+            ->map(function ($item) {
+                return $item->submission_id;
+            });
```

Comment:
Would here be the same to use `->pluck('submission_id')` instead of 
```
->select(['submission_id'])
->get()
->map(function ($item) {
return $item->submission_id;
})
```

Reply:
Yep!